layer/tracker/ui_progress_tracker.py

```diff
@@ -61,6 +62,9 @@ def track(self) -> Iterator["RunProgressTracker"]:
         with self._progress:
             self._init_tasks()
             yield self
+            time.sleep(
+                0.5
+            )  # Give some time so the latest metadata gets a chance to get rendered
```

Comment:
Is this a limitation with rich UI? Is there a better option we can build down the road that doesn't require sleeps?